invenio_rdm_records/services/access/service.py

```diff
@@ -500,32 +500,31 @@ def delete_grant(self, identity, id_, grant_id, uow=None):
 
         return True
 
-    def _exists(self, identity, created_by, record_id, request_type):
+    def _exists(self, created_by, record_id, request_type):
         """Return the request id if an open request already exists, else None."""
-        api_cls = current_requests_service.record_cls
-        model_cls = api_cls.model_cls
-
-        with db.session.no_autoflush:
-            requests = [
-                request
-                for request in (
-                    api_cls(rm.data, model=rm)
-                    for rm in model_cls.query.filter(
-                        model_cls.json["created_by"] == created_by,
-                        model_cls.json["topic"] == {"record": record_id},
-                        model_cls.json["type"].as_string() == request_type,
-                    )
-                    if not rm.is_deleted
-                )
-                if request.is_open
-            ]
-            if len(requests) > 1:
-                current_app.logger.error(
-                    f"Multiple access requests detected for: "
-                    f"record_pid{record_id}, creator: {created_by}"
-                )
-            if requests and len(requests) > 0:
-                return requests[0]
+        query_terms = [
+            dsl.Q("term", **{"topic.record": record_id}),
+            dsl.Q("term", **{"type": request_type}),
+            dsl.Q("term", **{"is_open": True}),
+        ]
+
+        # Build the query dynamically based on the keys in created_by
+        for key, value in created_by.items():
+            query_terms.append(dsl.Q("term", **{f"created_by.{key}": value}))
+
+        open_requests = current_requests_service.search(
+            system_identity,
```

Comment:
I have a bit of doubt about using `system_identity `for this, if we misuse _exists without being aware that it has superuser access we might expose accidentally some content. Especially because in line 527 you return the request item.

Reply:
We are still keeping the same behaviour, previously we were searching in the DB with no check on permissions. 
We only replaced it with OS to speed it up, it's a service method marked as "private" without passing the identity. It doesn't look easy to misuse this but I might be wrong, any suggestions on how to improve this are welcomed!

Note that this is called by anonymous users as well, meaning tha the identity will not be able to search (but we still need to check if there is a request in the system with the same email)

Reply:
Agreed with Nico that this has to use the `system_identity`, fixed the issue with `_exists` method in community_inclusion request below that should also use `system_indentity`